ambry-api/src/main/java/com/github/ambry/named/NamedBlobDb.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package com.github.ambry.named;
+
+import com.github.ambry.account.Container;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+public interface NamedBlobDb {
+  CompletableFuture<NamedBlobRecord> get(String accountName, String containerName, String blobName);
+
+  CompletableFuture<Page<NamedBlobRecord>> list(String accountName, String containerName, String blobNamePrefix, String continuationToken);
+
+  CompletableFuture<Void> put(NamedBlobRecord record);
```

Comment:
Optional: might be useful to return the NamedBlobRecord with blobId filled in.

Reply:
The `NamedBlobRecord` provided already contains the blob ID (since this method is called after Router.putBlob). I do agree that it may be a good practice to have some sort of response object where we can put extra metadata if needed on put/delete requests.